src/main/cpp/include/utils/vid_mapper_sql.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * The MIT License (MIT)
+ * Copyright (c) 2016 Intel Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+#ifndef GENOMICSDB_VID_MAPPER_SQL_H
+#define GENOMICSDB_VID_MAPPER_SQL_H
+
+#include "vid_mapper.h"
+#include <dbi/dbi.h>
+#include <string>
+#include <vector>
+#include <map>
+
+enum {
+  GENOMICSDB_VID_MAPPER_SUCCESS = 0x0,
+  GENOMICSDB_VID_MAPPER_FAILURE = 0x1
+};
+
+
+class SQLBasedVidMapper : public VidMapper {
+  public:
+    SQLBasedVidMapper();
+
+    int load_contig_info();
+
+    int load_metadata_from_db();
+
+    std::vector<ContigInfo> get_contigs() { return(m_contig_idx_to_info); }
+
+    std::vector<std::pair<int64_t, int>> get_contig_begin() {
+      return(m_contig_begin_2_idx);
+    }
+
+    std::vector<std::pair<int64_t, int>> get_contig_end() {
+      return(m_contig_end_2_idx);
+    }
+
+    ~SQLBasedVidMapper();
+  protected:
+    int num_contigs;
+    dbi_conn conn;
+    dbi_inst instance;
```

Comment:
Generally, we distinguish members of a class from local variables with a naming convention.
* Google coding style recommends adding "_" at the end of variable name.
* In GenomicsDB, I have been using the Hungarian notation - prefix every member variable name with "m_"
* Snake case for variable names.